src/CommunityToolkit.Maui/Converters/MathExpressionConverter/MathExpression.shared.cs

```diff
@@ -4,94 +4,179 @@
 
 namespace CommunityToolkit.Maui.Converters;
 
+enum MathTokenType
+{
+	Value,
+	Operator,
+};
+
+class MathToken
+{
+	public MathToken(MathTokenType type, string text, object? value)
+	{
+		Type = type;
+		Text = text;
+		Value = value;
+	}
+
+	public MathTokenType Type { get; }
+	public string Text { get; }
+	public object? Value { get; }
+};
```

Comment:
Let's use a record, this will give us the Equals and getHash implemented by value. Which is, probably, what we want here.

```suggestion
record MathToken (MathTokenType type, string text, object? value);
```

Reply:
Thanks @pictos. I implemented recorded as per your feedback.